sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/_ai_workspaces/capability_host.py

```diff
@@ -0,0 +1,226 @@
+        }
+        return cls(
+            base_path=cast(Dict, context[BASE_PATH_CONTEXT_KEY]),
+            **load_from_dict(CapabilityHostSchema, data, context, **kwargs),
+        )
+
+    @classmethod
+    def _from_rest_object(cls, rest_obj: RestCapabilityHost) -> "CapabilityHost":
+        """Convert a REST object into a CapabilityHost object.
+
+        :param cls: Indicates that this is a class method.
+        :type cls: class
+        :param rest_obj: The REST object to convert.
+        :type rest_obj: ~azure.ai.ml._restclient.v2024_10_01_preview.models._models_py3.CapabilityHost
+        :return: CapabilityHost object.
+        :rtype: ~azure.ai.ml.entities._workspace._ai_workspaces.capability_host.CapabilityHost
+        """
+        capability_host = CapabilityHost(
+            name=str(rest_obj.name),
+            description=(rest_obj.properties.description if rest_obj.properties else None),
+            ai_services_connections=(rest_obj.properties.ai_services_connections if rest_obj.properties else []),
+            storage_connections=(rest_obj.properties.storage_connections if rest_obj.properties else []),
+            vector_store_connections=(rest_obj.properties.vector_store_connections if rest_obj.properties else []),
+            capability_host_kind=(
+                rest_obj.properties.capability_host_kind if rest_obj.properties else CapabilityHostKind.AGENTS
+            ),
+        )
+        return capability_host
+
+    def _to_rest_object_for_hub(self) -> RestCapabilityHost:
+        """
+        Convert the CapabilityHost instance to a RestCapabilityHost object for a Hub workspace.
+
+        :return: A RestCapabilityHost object representing the capability host for a Hub workspace.
+        :rtype: azure.ai.ml._restclient.v2024_10_01_preview.models._models_py3.CapabilityHost
+        """
+
+        properties = RestCapabilityHostProperties(
+            description=self.description, capability_host_kind=self.capability_host_kind
+        )
+        resource = RestCapabilityHost(
+            properties=properties,
+        )
+        return resource
+
+    def _to_rest_object_for_project(self) -> RestCapabilityHost:
+        """
+        Convert the CapabilityHost instance to a RestCapabilityHost object for a Project workspace.
+
+        :return: A RestCapabilityHost object representing the capability host for a Hub workspace.
+        :rtype: azure.ai.ml._restclient.v2024_10_01_preview.models._models_py3.CapabilityHost
+        """
+
+        properties = RestCapabilityHostProperties(
+            ai_services_connections=self.ai_services_connections,
+            storage_connections=self.storage_connections,
+            vector_store_connections=self.vector_store_connections,
+            description=self.description,
+            capability_host_kind=self.capability_host_kind,
+        )
+        resource = RestCapabilityHost(
+            properties=properties,
+        )
+        return resource
```

Comment:
what's the difference between creating for hub/project? And if they should be different, let's still use one function instead of 2

Reply:
agree and done